src/module.c

```diff
@@ -1566,59 +1578,73 @@ int moduleReplyWithCollection(RedisModuleCtx *ctx, long len, int type) {
         case COLLECTION_REPLY_ATTRIBUTE:
             addReplyAttributeLen(c,len);
             break;
-        case COLLECTION_REPLY_PUSH:
-            addReplyPushLen(c,len);
-            break;
         default:
             serverPanic("Invalid module reply type %d", type);
         }
     }
     return REDISMODULE_OK;
 }
 
-
-/* Reply with an array type of 'len' elements. However 'len' other calls
+/* --------------------------------------------------------------------------
+ * ## Reply with collection functions
+ *
+ * Collection types include: Array, Map, Set and Attribute.
```

Comment:
this text will appear in the reference docs of `RedisModule_ReplyWithArray` see:
https://redis.io/topics/modules-api-ref#RedisModule_ReplyWithArray

i'm not sure if it's wise to add a general documentation here.
maybe instead it can remain specific for arrays, and just have the other collection methods less verbose, possibly referring to RM_ReplyWithArray

@zuiderkwast WDYT?

Reply:
@ashtul I think these functions should belong to the ["Reply APIs"](https://redis.io/topics/modules-api-ref#reply-apis) section. I don't like a separate section for collection reply functions. We can add this text for collections in the Reply APIs section introduction.

@oranagra This comment block is separated from the comment block above RM_ReplyWithArray, so it will not be part of the docs for that function.